src/dwfl_module.cc

```diff
@@ -5,90 +5,76 @@
 
 #include "dwfl_module.hpp"
 
+#include "ddres.hpp"
 #include "logger.hpp"
 
 namespace ddprof {
 
-static const char *get_filename(const char *full_path) {
-  const char *right_slash = strrchr(full_path, '/');
-  if (!right_slash) {
-    return full_path;
+DDRes update_module(Dwfl *dwfl, ProcessAddress_t pc,
+                    const DDProfModRange &mod_range,
+                    const FileInfoValue &fileInfoValue, DDProfMod &ddprof_mod) {
+  const std::string &filepath = fileInfoValue.get_path();
+  const char *module_name = strrchr(filepath.c_str(), '/') + 1;
+  if (fileInfoValue._errored) { // avoid bouncing on errors
+    LG_DBG("DSO Previously errored - mod (%s)", module_name);
+    return ddres_warn(DD_WHAT_MODULE);
   }
-  return right_slash + 1;
-}
-
-DDProfMod update_module(Dwfl *dwfl, ProcessAddress_t pc, const Dso &dso,
-                        const FileInfoValue &fileInfoValue) {
-  if (!dwfl)
-    return DDProfMod();
 
-  if (fileInfoValue._errored) {
-    LG_DBG("DSO Previously errored - mod (%s)", dso._filename.c_str());
-    return DDProfMod();
-  }
   // Now that we've confirmed a separate lookup for the DSO based on procfs
   // and/or perf_event_open() "extra" events, we do two things
   // 1. Check that dwfl has a cache for this PID/pc combination
   // 2. Check that the given cache is accurate to the DSO
-
-  DDProfMod ddprof_mod;
-
   ddprof_mod._mod = dwfl_addrmodule(dwfl, pc);
 
   if (ddprof_mod._mod) {
     const char *main_name = nullptr;
     dwfl_module_info(ddprof_mod._mod, 0, &ddprof_mod._low_addr,
                      &ddprof_mod._high_addr, 0, 0, &main_name, 0);
-    if (ddprof_mod._low_addr != dso._start - dso._pgoff) {
-      // not using only addresses (as we get off by 1 page errors)
-      const std::string &filepath = fileInfoValue.get_path();
-      if (filepath.empty() || main_name == nullptr) {
-        return DDProfMod();
-      } else {
-        // This check is slightly fragile (checking file names)
-        // We are not comparing the full path as we use the proc maps
-        // to access them. So the root path could change while
-        // this would still be the same file.
-        const char *dso_name = get_filename(filepath.c_str());
-        const char *mod_name = get_filename(main_name);
-        if (strcmp(mod_name, dso_name) != 0) {
-          // A dso replaced the mod - flag as incoherent for future release
-          LG_NTC("Incoherent DSO (%s) %lx != %lx dwfl_module (%s)",
-                 filepath.c_str(), dso._start - dso._pgoff,
-                 ddprof_mod._low_addr, main_name);
-          return DDProfMod(DDProfMod::kInconsistent);
-        }
-      }
+    if (ddprof_mod._low_addr != mod_range._low_addr) {
+      LG_NTC("Incoherent Modules (%s-%s) %lx != %lx dwfl_module)", module_name,
```

Comment:
Test that when a new dso is mapped (within an existing mod_range), we update the ddprof_module

Reply:
We don't, follow up ticket.